bin/proving-service/src/proxy/status.rs

```diff
@@ -1,103 +1,226 @@
-use std::sync::Arc;
-
-use pingora::{
-    apps::HttpServerApp,
-    http::ResponseHeader,
-    prelude::*,
-    protocols::{Stream, http::ServerSession},
-    server::ShutdownWatch,
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use pingora::{server::ListenFds, services::Service};
+use tokio::{net::TcpListener, sync::watch, time::interval};
+use tokio_stream::wrappers::TcpListenerStream;
+use tonic::{Request, Response, Status, transport::Server};
+use tracing::{error, info};
+
+use super::worker::WorkerHealthStatus as RustWorkerHealthStatus;
+use crate::{
+    commands::PROXY_HOST,
+    generated::{
+        proving_service::ProofType,
+        proxy_status::{
+            ProxyStatusRequest, ProxyStatusResponse, WorkerHealthStatus, WorkerStatus,
+            proxy_status_api_server::{ProxyStatusApi, ProxyStatusApiServer},
+        },
+    },
+    proxy::LoadBalancerState,
 };
-use serde::Serialize;
-use tonic::async_trait;
-use tracing::error;
-
-use super::worker::WorkerHealthStatus;
-use crate::{commands::worker::ProverType, proxy::LoadBalancerState};
-
-// Status of a worker
-#[derive(Debug, Serialize)]
-pub struct WorkerStatus {
-    address: String,
-    version: String,
-    status: WorkerHealthStatus,
-}
 
-/// Status of the proxy
-#[derive(Debug, Serialize)]
-pub struct ProxyStatus {
-    version: String,
-    prover_type: ProverType,
-    workers: Vec<WorkerStatus>,
-}
+/// Update status every 5 seconds
+const STATUS_UPDATE_INTERVAL_SECS: u64 = 5;
```

Comment:
I think this should come through as a Clap arg with env var support. Fed through into the service and updater. WDYT?

Would need to add this to `bin/proving-service/.env` also.

Reply:
Also maybe a default of 10 or 20s might be better? The only thing updated is the worker count / health right?

Reply:
Yep, sounds good 👍🏼 
> The only thing updated is the worker count / health right?
It is, yes.

Reply:
@SantiagoPittella please be aware of #998. If it merges before this PR, we should update yours to use humantime for this field. If you merge before it, I'll try to remember to do a followup issue!

Reply:
Saw that #998 was merged, i saw that most of the remote prover vars weren't updated, will check other binaries too and address it in another PR. I created #1005